### PR TITLE
Add container task runner type, move fully to github actions, go.mod to 1.17

### PR DESCRIPTION
**Pull request by fishnix**
* Add the container task runner
* Remove the circleci config, add standardized github action workflow
* Bump the go.mod to 1.17
* Add time tracking to the loader, scheduler and executer loops

I'm wondering if we can make those config endpoint templates less cumbersome (even though they will eventually go into api-config).  There's a lot of repeat text for each of the runners and the static endpoints are probably less useful than I thought they would be initially.  In addition, there should be a Validate() func on the runner interface that should be executed when a job is created or updated to be sure we have a valid job in the repository.  I can open some jira stories if that sounds good.

### CODE REVIEW

File: config/config.example.json [hunk 1]
```diff
@@ -25,6 +25,15 @@
         "auth_header": "X-Auth-Token",
         "token": "zzzzz",
         "encrypt_token": true
+      },
```
Comment (tenyo) [inline config/config.example.json:28]: We also need to update the deco file

File: jobs/taskrunner_test.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+				config: map[string]interface{}{
+					"endpoint":         "http://127.0.0.1:8080/v1/ecs/acct1/cluster/clu1/taskdefs/td1/tasks",
+					"endpointTemplate": "http://127.0.0.1:8080/v1/ecs/{{.Account}}/cluster/{{.Cluster}}/taskdefs/{{ .Name }}/tasks",
+				},
+			},
+			want: &TaskRunner{
+				AuthHeader:       "X-Auth-Token",
+				Encrypt:          true,
+				Endpoint:         "http://127.0.0.1:8080/v1/ecs/acct1/cluster/clu1/taskdefs/td1/tasks",
+				EndpointTemplate: "http://127.0.0.1:8080/v1/ecs/{{.Account}}/cluster/{{.Cluster}}/taskdefs/{{ .Name }}/tasks",
```
Comment (tenyo) [inline jobs/taskrunner_test.go:80]: It looks like if both of these are set we should only use `Endpoint`, but we are still setting both?

File: jobs/taskrunner.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	Account  string
+	Cluster  string
+	Name     string
+	count    int
+	endpoint string
+}
+
+// NewTaskRunner creates and configures a new task runner.  An endpoint or endpoint template is required
+// the endpoint and endpoint template are not currently validated but this can/should be done in the future.  If a
+// a token is passed, it will be configured but is not required.
```
Comment (tenyo) [inline jobs/taskrunner.go:36]: ✂️ `a`

**Review by tenyo — COMMENTED** [review]
Agree that `Validate()` will be useful.
Not sure what you mean about the endpoint templates, it's just a single string, right?

File: config/config.example.json (hunk 1 above)
Comment (fishnix) [reply config/config.example.json:28]: good call

File: jobs/taskrunner_test.go (hunk 2 above)
Comment (fishnix) [reply jobs/taskrunner_test.go:80]: derp - bad test name.  the endpoint and endpointTemplate will both get set, its up to the `Run()` to decide how it gets used.  that's partly why I think we need the `Validate()`. Even if different runners have different requirements, they should at least make sure what's set up will work.

**Comment by fishnix** [comment]
> Not sure what you mean about the endpoint templates, it's just a single string, right?

Yeah but there might be a better way to structure the config so we don't keep repeating that very similar string:

```json
"endpointTemplate": "http://127.0.0.1:8080/v1/ec2/{{.Account}}/instances/{{.InstanceID}}/power",
"endpointTemplate": "http://127.0.0.1:8080/v1/ecs/{{.Account}}/clusters/{{.Cluster}}/services/{{.Name}}",
"endpointTemplate": "http://127.0.0.1:8080/v1/ecs/{{.Account}}/cluster/{{.Cluster}}/taskdefs/{{ .Name }}/tasks",
```

maybe it's not possible without completely refactoring the config though....

File: jobs/taskrunner.go [hunk 4]
```diff
@@ -32,8 +32,8 @@ type TaskRunnerRunInput struct {
 }
 
 // NewTaskRunner creates and configures a new task runner.  An endpoint or endpoint template is required
-// the endpoint and endpoint template are not currently validated but this can/should be done in the future.  If a
-// a token is passed, it will be configured but is not required.
+// the endpoint and endpoint template are not currently validated but this can/should be done in the future.
+// If a token is passed, it will be configured but is not required.a
```
Comment (tenyo) [inline jobs/taskrunner.go:36]: 👀 the jumping `a` 😂

Comment (fishnix) [reply jobs/taskrunner.go:36]: oh jeebus 😆

**Comment by tenyo** [comment]
Looks good 👍

**Pull request merged**